worlds/civ_6/__init__.py

```diff
@@ -0,0 +1,204 @@
+
+            self.multiworld.itempool += [self.create_item(
+                item_to_create)]
+            if item.civ_name in early_items:
+                self.multiworld.early_items[self.player][item_to_create] = 1
+            elif self.item_table[item_name].era in [EraType.ERA_ATOMIC, EraType.ERA_INFORMATION, EraType.ERA_FUTURE]:
+                for location in early_locations:
+                    found_location = None
+                    try:
+                        found_location = self.get_location(location.name)
+                        forbid_item(found_location, item_to_create, self.player)
+                    except KeyError:
+                        pass
+
+        # Era items
+        if self.options.progression_style == "eras_and_districts":
+            # Add one less than the total number of eras (start in ancient, don't need to find it)
+            for era in EraType:
+                if era.value == "ERA_ANCIENT":
+                    continue
+                progressive_era_item = self.item_table.get("Progressive Era")
+                assert progressive_era_item is not None
+                self.multiworld.itempool += [self.create_item(progressive_era_item.name)]
+
+            self.multiworld.early_items[self.player]["Progressive Era"] = 2
+
+        num_filler_items = 0
+        # Goody items, create 10 by default if options are enabled
+        if self.options.shuffle_goody_hut_rewards:
+            num_filler_items += 10
+
+        if self.options.boostsanity:
+            num_filler_items += len(get_boosts_data())
+
+        filler_count = {rarity: math.ceil(FILLER_DISTRIBUTION[rarity] * num_filler_items) for rarity in FillerItemRarity.__reversed__()}
+        filler_count[FillerItemRarity.COMMON] -= sum(filler_count.values()) - num_filler_items
+        self.multiworld.itempool += [
+            self.create_item(get_random_filler_by_rarity(self, rarity).name)
+            for rarity, count in filler_count.items()
+            for _ in range(count)
+        ]
+
+    def post_fill(self) -> None:
+        if self.options.pre_hint_items == "none":
+            return
+
+        show_flags = {
+            ItemClassification.progression: self.options.pre_hint_items != "none",
+            ItemClassification.useful: self.options.pre_hint_items == "no_junk" or self.options.pre_hint_items == "all",
+            ItemClassification.filler: self.options.pre_hint_items == "all",
+        }
+
+        start_location_hints: Set[str] = self.options.start_location_hints.value
+        for location_name, location_data in self.location_table.items():
+            if location_data.location_type != CivVICheckType.CIVIC and location_data.location_type != CivVICheckType.TECH:
+                continue
+
+            location: CivVILocation = self.get_location(location_name)  # type: ignore
+
+            if not location.item or not show_flags.get(location.item.classification, False):

```

Comment:
I don't think this properly handles Progression+Useful items. Speaking of which, have you considered if you'd want to mark the especially useful Progression items in your game as Progression+Useful?

Reply:
Didn't know about especially useful flagging, however I don't think it'll make too big of a difference right now since there are so many different ways to win 🤷